### ISSUE TITLE
IO#seek and tell do not properly handle offsets larger than int32

### ISSUE THREAD

**Issue by headius** [jruby/jruby#3817]
### Environment

JRuby 9.1

### Expected Behavior

`IO#seek` and `IO#tell` should handle offsets larger than int32 range

### Actual Behavior

As seen in #3435, we were not handling the return value of `lseek` properly, interpreting all negative values as error rather than just -1. That was fixed, but the larger issue is that our native `lseek` and much of the code that calls it only supports int32 range.

I have a commit coming shortly to fix the main #3435 issue, but this large problem needs to be addressed after 9.1.

**Comment by headius** [comment]
See #3664 for another report partially fixed by #3435.

See jnr/jnr-posix#71 for a PR to fix at least some of these APIs in jnr-posix.

**Comment by headius** [comment]
I had some trouble figuring out how to test this.

**Comment by byteit101** [comment]
I had a small (well, not file size small, but code small) test case when I reported this on https://github.com/jruby/jruby/issues/3664 that could be potentially adapted as a unit test provided you have 2G/5G of disk space

**Comment by headius** [comment]
@byteit101 Yeah it's the file size I'm concerned about. I was able to use /dev/zero to test the jnr-posix fix, but that seems to blow up for me with ruby.

**Comment by headius** [comment]
Oh nevermind...I had reversed seek params. Test coming.

**Comment by Goltergaul** [comment]
Hi is this issue fixed now on newer jrubies?

**Comment by felixbuenemann** [comment]
This is still broken in `StringIO` on jruby 9.2.9.0:

```ruby
s = StringIO.new
s.seek(2**31-1)
=> 0
s.seek(2**31)
RangeError (integer 2147483648 too big to convert to `int')
```
